vm/safepoints.cpp

```diff
@@ -39,6 +40,14 @@ void safepoint_state::handle_safepoint(factor_vm* parent, cell pc) volatile {
     if (atomic::load(&parent->sampling_profiler_p))
       parent->end_sampling_profiler();
     std::cout << "Interrupted\n";
+    if (atomic::load(&ctrl_break_p) != 0) {
+      // Ctrl-Break throws an exception, interrupting the main thread, same
+      // as the "t" command in the factorbug debugger. But for Ctrl-Break to
+      // work we don't require the debugger to be activated, or even enabled.
+      atomic::store(&fep_p, false);
+      parent->general_error(ERROR_INTERRUPT, false_object, false_object);
+      FACTOR_ASSERT(false);
+    }
```

Comment:
Changing the signature of `enqueue_fep` I don't think is right. Instead add a flag to the `factor_vm` struct: `volatile cell skip_debugger_p` which you set to true in `primitive_enable_ctrl_break`. Then dispatch on that:

```c++
if (atomic::load(&parent->skip_debugger)) { parent->general_error ... } 
else { parent->factorbug() ... } 
```